### PR TITLE
Add bison

### PR DESCRIPTION
**Pull request by jakirkham**
Builds bison from source for Linux and Mac. Borrowed from conda recipes.

### CODE REVIEW

File: recipes/bison/build.sh [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
+
+./configure --prefix="$PREFIX"
+make
+make install
```
Comment (ocefpaf) [inline recipes/bison/build.sh]: Is there a `make check` defined for bison?

File: recipes/bison/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  version: {{ version }}
+
+source:
+  fn: bison-{{ version }}.tar.gz
+  url: http://ftp.gnu.org/gnu/bison/bison-{{ version }}.tar.gz
+  md5: a586e11cd4aff49c3ff6d3b6a4c9ccf8
+
+build:
+  skip: true  # [win]
+  number: 1
```
Comment (ocefpaf) [inline recipes/bison/meta.yaml]: We are starting from `0`. (Sorry, python bias :wink:)

File: recipes/bison/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  build:
+    - m4
+  
+  run:
+    - m4
+
+about:
+    home: https://www.gnu.org/software/bison/
+    license: GPL 3
+
```
Comment (ocefpaf) [inline recipes/bison/meta.yaml]: The linter will complain about the missing `test` and `summary` fields.

File: recipes/bison/meta.yaml (hunk 2 above)
Comment (jakirkham) [inline recipes/bison/meta.yaml]: Ah, sorry, came from the copied over recipe so was trying to be consistent. Will switch to `0`.

File: recipes/bison/build.sh (hunk 1 above)
Comment (jakirkham) [inline recipes/bison/build.sh]: Good call. Will add this.

**Comment by jakirkham** [comment]
@jdreaver @ccordoba12, would you be interested in being added as maintainers?

File: recipes/bison/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+extra:
+    recipe-maintainers:
+        - jakirkham
```
Comment (ocefpaf) [inline recipes/bison/meta.yaml]: Feel free to add me as maintainer to any build tool you submit to conda-forge.

File: recipes/bison/meta.yaml (hunk 3 above)
Comment (jakirkham) [inline recipes/bison/meta.yaml]: Definitely. Sorry, forgot summary. Will add.

For a test, thinking some simple `bison --help` command.

File: recipes/bison/meta.yaml (hunk 4 above)
Comment (jakirkham) [inline recipes/bison/meta.yaml]: Ok, sure, will do. Thanks.

**Comment by jakirkham** [comment]
Also, `bison` includes `yacc`, but I can see value in having a separate `yacc` recipe.

File: recipes/bison/meta.yaml (hunk 3 above)
Comment (ocefpaf) [inline recipes/bison/meta.yaml]: > For a test, thinking some simple bison --help command.

:+1:

**Comment by jakirkham** [comment]
Made the recommended changes. How is this looking?

**Comment by jjhelmus** [comment]
FYI all of these build tool packages will break if we move to a CentOS 5 based docker image as they are linking against the CentOS 6 based glibc.

**Comment by jakirkham** [comment]
Looks like CI is failing on Circle. Not sure why though. :/

**Comment by jakirkham** [comment]
> FYI all of these build tool packages will break if we move to a CentOS 5 based docker image as they are linking against the CentOS 6 based glibc.

Sure. Wouldn't we be able to just rebuild them or am I missing something here?

**Comment by jjhelmus** [comment]
> Sure. Wouldn't we be able to just rebuild them or am I missing something here?

Yes, but they would need to be rebuilt in the correct dependency order.  Just wanted to bring it up before we go along too far.

**Comment by jakirkham** [comment]
Ok, cool. What is the status on CentOS 5 docker container? Is there some work needed for it? Happy to help in any way that I can.

**Comment by jjhelmus** [comment]
I think there is a version we should merge that has the various X11 libraries and some build tool, we will want to re-visit what all should be included in the image but I think for the time being a quick fix is best.  I need a bit of time to work on this and currently have some "real job" tasks that need my attention.

**Comment by ocefpaf** [comment]
Changing the docker image will be a nice exercise for the infrastructure :wink: 

PS: The failure is with the `make check`.  I am not sure if you uncovered a bug or if it is a failure due to our build system.

**Comment by jakirkham** [comment]
> Changing the docker image will be a nice exercise for the infrastructure :wink:

Definitely.

> PS: The failure is with the `make check`. I am not sure if you uncovered a bug or if it is a failure due to our build system.

It is both a bug for bison and a issue I need to fix. ( http://savannah.gnu.org/support/?108352 ) Basically need to add flex, which I also have so will do that first and then we can come back to this. Unless we already have flex.

**Comment by jakirkham** [comment]
Flex was added here ( https://github.com/conda-forge/staged-recipes/pull/120 ).

**Comment by pelson** [comment]
Added a WIP to the title. @jakirkham, any idea on where to take this next?

**Comment by jakirkham** [comment]
This is backlogged until we can get `flex` to work as it is required to build `bison`.

**Comment by jakirkham** [comment]
BTW, `flex` got moved to this PR ( https://github.com/conda-forge/staged-recipes/pull/121 ).

**Comment by jakirkham** [comment]
Alright, this is all fixed up with tests passing. If there is no more feedback, this is ready to merge. :tada:

**Comment by jakirkham** [comment]
Alright, I'm just going to merge this. Eager to see this out in the wild after two (plus) weeks. Feedback welcome at the feedstock. Thanks everyone. :smile:

**Comment by ocefpaf** [comment]
Thanks @jakirkham!

**Pull request merged**